plugin/src/main/java/org/opensearch/ml/action/mcpserver/McpStatelessServerHolder.java

```diff
@@ -0,0 +1,169 @@
+import org.opensearch.ml.common.transport.mcpserver.requests.register.McpToolRegisterInput;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.client.Client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.modelcontextprotocol.server.McpServer;
+import io.modelcontextprotocol.server.McpStatelessAsyncServer;
+import io.modelcontextprotocol.spec.McpSchema;
+import lombok.extern.log4j.Log4j2;
+import reactor.core.publisher.Mono;
+
+@Log4j2
+public class McpStatelessServerHolder {
+
+    private final McpToolsHelper mcpToolsHelper;
+    private final Client client;
+    private final ThreadPool threadPool;
+    private static final int SYNC_MCP_TOOLS_JOB_INTERVAL = 10;
+    public static Map<String, Long> IN_MEMORY_MCP_TOOLS = new ConcurrentHashMap<>();
+    private static volatile McpStatelessAsyncServer mcpStatelessAsyncServer;
+    private static volatile OpenSearchMcpStatelessServerTransportProvider mcpStatelessServerTransportProvider;
+    private static volatile Boolean initialized = false;
+
+    public McpStatelessServerHolder(McpToolsHelper mcpToolsHelper, Client client, ThreadPool threadPool) {
+        this.mcpToolsHelper = mcpToolsHelper;
+        this.client = client;
+        this.threadPool = threadPool;
+    }
+
+    public void initialize() {
+        if (initialized) {
+            return;
+        }
+        synchronized (McpStatelessServerHolder.class) {
+            if (initialized) {
+                return;
+            }
+            try {
+                mcpStatelessServerTransportProvider = new OpenSearchMcpStatelessServerTransportProvider(new ObjectMapper());
+
+                McpSchema.ServerCapabilities serverCapabilities = McpSchema.ServerCapabilities
+                    .builder()
+                    .tools(true)
+                    .logging()
+                    .resources(false, false)  // We don't support resources
+                    .prompts(false)           // We don't support prompts
+                    .build();
+
+                log.info("Building MCP server without pre-loaded tools (dynamic loading)...");
+                mcpStatelessAsyncServer = McpServer
+                    .async(mcpStatelessServerTransportProvider)
+                    .serverInfo("OpenSearch-MCP-Stateless-Server", "0.1.0")
+                    .capabilities(serverCapabilities)
+                    .instructions("OpenSearch MCP Stateless Server - provides access to ML tools without sessions")
+                    .build();
+
+                log.info("Stateless MCP server created successfully");
+
+                autoLoadAllMcpTools(
+                    ActionListener
+                        .wrap(
+                            success -> log.info("Initial tool loading completed successfully"),
+                            error -> log.error("Initial tool loading failed", error)
+                        )
+                );
+                initialized = true;
```

Comment:
I think we should move this line inside `autoLoadAllMcpTools` success branch ? If load MCP tools failed, we should not set `initialized` as `true` right?